lib/Gedmo/Tree/Strategy/ORM/Nested.php

```diff
@@ -534,6 +534,9 @@ public function shiftRL(EntityManager $em, $class, $first, $delta, $root = null)
                 if ($node instanceof Proxy && !$node->__isInitialized__) {
                     continue;
                 }
+                if (get_class($node) !== $class) {
```

Comment:
tests are failing, maybe you could use `$meta->getReflectionClass()->hasProperty($config['left'])` ?